crates/nu-command/src/system/inspect.rs

```diff
@@ -0,0 +1,369 @@
+use nu_protocol::{
+    Category, Example, IntoInterruptiblePipelineData, PipelineData, ShellError, Signature, Span,
+    SyntaxShape, Type, Value,
+};
+
+#[derive(Clone)]
+pub struct Inspect;
+
+impl Command for Inspect {
+    fn name(&self) -> &str {
+        "inspect"
+    }
+
+    fn usage(&self) -> &str {
+        "Inspect closure contents."
+    }
+
+    fn signature(&self) -> nu_protocol::Signature {
+        Signature::build("inspect")
+            .required(
+                "closure",
+                SyntaxShape::Closure(Some(vec![SyntaxShape::Any])),
+                "the closure to run",
+            )
+            .input_output_types(vec![(Type::Any, Type::Any), (Type::Nothing, Type::Any)])
+            .allow_variants_without_examples(true)
+            .category(Category::System)
+    }
+
+    fn run(
+        &self,
+        engine_state: &EngineState,
+        stack: &mut Stack,
+        call: &Call,
+        input: PipelineData,
+    ) -> Result<nu_protocol::PipelineData, nu_protocol::ShellError> {
+        // This was all delightfully stolen from benchmark :)
+        let capture_block: Closure = call.req(engine_state, stack, 0)?;
+        let block = engine_state.get_block(capture_block.block_id);
+        let ctrlc = engine_state.ctrlc.clone();
+
+        // let redirect_stdout = call.redirect_stdout;
+        // let redirect_stderr = call.redirect_stderr;
+
+        let mut stack = stack.captures_to_stack(&capture_block.captures);
+
+        // In order to provide the pipeline as a positional, it must be converted into a value.
+        // But because pipelines do not have Clone, this one has to be cloned as a value
+        // and then converted back into a pipeline for eval_block().
+        // So, the metadata must be saved here and restored at that point.
+        // let input_metadata = input.metadata();
+        let input_val = input.into_value(call.head);
+
+        if let Some(var) = block.signature.get_positional(0) {
+            if let Some(var_id) = &var.var_id {
+                stack.add_var(*var_id, input_val);
+            }
+        }
+
+        let elements = get_pipeline_elements(engine_state, &mut stack, block)?;
+
+        // Leaving these comments here because at some point we'd like to be able to
+        // run commands in the closure and instrument them for more info.
+
+        // Get the start time after all other computation has been done.
+        // let start_time = Instant::now();
+        // eval_block(
+        //     engine_state,
+        //     &mut stack,
+        //     block,
+        //     input_val.into_pipeline_data_with_metadata(input_metadata),
+        //     redirect_stdout,
+        //     redirect_stderr,
+        // )?
+        // .into_value(call.head);
+
+        // let end_time = Instant::now();
+
+        // let output = Value::Duration {
+        //     val: (end_time - start_time).as_nanos() as i64,
+        //     span: call.head,
+        // };
```

Comment:
At one point you will gonna drive me crazy with the commented out code :P

Reply:
🤣 Ya, I was leaving this in there for a time because I wasn't sure if this command was going to do `eval_block` or not. I'm not sure how I'd do it now considering the other things this command does so I'll probably end up dropping that idea.